File: common/public/src/main/java/dcp/mc/pstp/config/Revival.java
package dcp.mc.pstp.config;

import org.jetbrains.annotations.NotNull;

public record Revival(
        boolean enabled,
        boolean named,
        @NotNull String @NotNull [] blocks,
        @NotNull String @NotNull [] blacklist,
        @NotNull String @NotNull [] whitelist
)
{
    Revival() {
        this(true, true, new String[] { "minecraft:copper_block" }, new String[0], new String[0]);
    }
}

File: common/src/main/kotlin/dcp/mc/pstp/config/Revival.kt
package dcp.mc.pstp.config

@JvmRecord
data class Revival(
    val enabled: Boolean = true,
    val named: Boolean = true,
    val blocks: List<String> = listOf("minecraft:copper_block"),
    val blacklist: List<String> = emptyList(),
    val whitelist: List<String> = emptyList(),
)